contacts/src/main/java/com/maxrzhe/contacts/repository/ContactRepoWithRestApi.kt

```diff
@@ -0,0 +1,120 @@
+package com.maxrzhe.contacts.repository
+
+import android.app.Application
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.maxrzhe.contacts.app.ContactsApp
+import com.maxrzhe.contacts.data.ContactListResponse
+import com.maxrzhe.contacts.model.ContactMapping
+import com.maxrzhe.core.model.Contact
+
+class ContactRepoWithRestApi private constructor(app: Application, type: RepositoryType) :
+    Repository {
+
+    private val contactsApi = (app as ContactsApp).contactsApi
+    private val repository = RepositoryFactory.create(app, type)
+    private val allContactsList: MutableLiveData<List<Contact.Existing>> = MutableLiveData()
+
+    override suspend fun findById(fbId: String): Contact.Existing? {
+        return null
+    }
+
+    override suspend fun add(contact: Contact.New) {
+        val response = contactsApi.postContactAsync(contact).await()
+        if (response.name.isNotEmpty()) {
+            with(contact) {
+                val newContact = Contact.New(
+                    fbId = response.name,
+                    name = name,
+                    email = email,
+                    phone = phone,
+                    image = image,
+                    birthDate = birthDate,
+                    isFavorite = isFavorite
+                )
+                repository.add(newContact)
+            }
+        }
+    }
+
+    override suspend fun addAll(contacts: List<Contact.New>) {
+        repository.addAll(contacts)
+    }
+
+    override suspend fun update(contact: Contact.Existing) {
+        repository.update(contact)
+    }
+
+    override suspend fun updateAll(contacts: List<Contact.Existing>) {
+        repository.updateAll(contacts)
+    }
+
+    override suspend fun delete(contact: Contact.Existing) {
+        repository.delete(contact)
+    }
+
+    override suspend fun deleteAll(contacts: List<Contact.Existing>) {
+        repository.deleteAll(contacts)
+    }
+
+
+    override suspend fun findAll(): LiveData<List<Contact.Existing>> {
+        val dbContacts: LiveData<List<Contact.Existing>> = repository.findAll()
+        allContactsList.value = dbContacts.value
+        val apiContacts: ContactListResponse = contactsApi.getContactsListAsync().await()
+
+        val contacts = dbContacts.value
+        val existing = apiContacts.mapNotNull {
+            val oldContact = exist(it.key, contacts)
+            if (oldContact != null) {
```

Comment:
oldContact?.let{ContactMapping.contactRestToContactExisting(it.key, it.value, oldContact)}